src/test/java/org/threeten/extra/chrono/TestInternationalFixedChronology.java

```diff
@@ -0,0 +1,794 @@
+        InternationalFixedDate date = InternationalFixedDate.of (2000, 1, 4);
+        date.with (Month.APRIL);
+    }
+
+    //-----------------------------------------------------------------------
+    // LocalDate.with(InternationalFixedDate)
+    //-----------------------------------------------------------------------
+    @Test
+    public void test_LocalDate_adjustToInternationalFixedDate () {
+        InternationalFixedDate date = InternationalFixedDate.of (2012, 6, 15);
+        LocalDate test = LocalDate.MIN.with (date);
+        assertEquals (test, LocalDate.of (2012, 6, 4));
+    }
+
+    @Test
+    public void test_LocalDateTime_adjustToInternationalFixedDate () {
+        InternationalFixedDate date = InternationalFixedDate.of (2012, 6, 15);
+        LocalDateTime test = LocalDateTime.MIN.with (date);
+        assertEquals (test, LocalDateTime.of (2012, 6, 4, 0, 0));
+    }
+
+    //-----------------------------------------------------------------------
+    // InternationalFixedDate.plus
+    //-----------------------------------------------------------------------
+    @DataProvider (name = "plus")
+    Object[][] data_plus () {
+        return new Object[][] {
+                { 2014, 5, 26, 0, ChronoUnit.DAYS, 2014, 5, 26 },
+                { 2014, 5, 26, 8, ChronoUnit.DAYS, 2014, 6, 6 },
+                { 2014, 5, 26, -3, ChronoUnit.DAYS, 2014, 5, 23 },
+                { 2014, 5, 26, 0, ChronoUnit.WEEKS, 2014, 5, 26 },
+                { 2014, 5, 26, 3, ChronoUnit.WEEKS, 2014, 6, 19 },
+                { 2014, 5, 26, -5, ChronoUnit.WEEKS, 2014, 4, 19 },
+                { 2014, 5, 26, 0, ChronoUnit.MONTHS, 2014, 5, 26 },
+                { 2014, 5, 26, 3, ChronoUnit.MONTHS, 2014, 8, 26 },
+                { 2014, 5, 26, -5, ChronoUnit.MONTHS, 2013, 13, 26 },
+                { 2014, 5, 26, 0, ChronoUnit.YEARS, 2014, 5, 26 },
+                { 2014, 5, 26, 3, ChronoUnit.YEARS, 2017, 5, 26 },
+                { 2014, 5, 26, -5, ChronoUnit.YEARS, 2009, 5, 26 },
+                { 2014, 5, 26, 0, ChronoUnit.DECADES, 2014, 5, 26 },
+                { 2014, 5, 26, 3, ChronoUnit.DECADES, 2044, 5, 26 },
+                { 2014, 5, 26, -5, ChronoUnit.DECADES, 1964, 5, 26 },
+                { 2014, 5, 26, 0, ChronoUnit.CENTURIES, 2014, 5, 26 },
+                { 2014, 5, 26, 3, ChronoUnit.CENTURIES, 2314, 5, 26 },
+                { 2014, 5, 26, -5, ChronoUnit.CENTURIES, 1514, 5, 26 },
+                { 2014, 5, 26, 0, ChronoUnit.MILLENNIA, 2014, 5, 26 },
+                { 2014, 5, 26, 3, ChronoUnit.MILLENNIA, 5014, 5, 26 },
+
+                { 2012, 13, 6, 3, ChronoUnit.MONTHS, 2013, 3, 6 },
+                { 2011, 13, 26, 1, ChronoUnit.YEARS, 2012, 13, 26 },
+                { 2014, 13, 26, -2, ChronoUnit.YEARS, 2012, 13, 26 },
+                { 2012, 13, 26, -6, ChronoUnit.YEARS, 2006, 13, 26 },
+                { 2012, 13, 6, -6, ChronoUnit.YEARS, 2006, 13, 6 },
+        };
+    }
+
+    @DataProvider (name = "plusLeap")
```

Comment:
You may as well test the year-day here, too (I originally did these separate data constructors because dealing with leap stuff isn't bi-directional).  Except you're not testing leap-day stuff anyways...